pkg/operator/ceph/file/controller.go

```diff
@@ -222,9 +222,11 @@ func (r *ReconcileCephFilesystem) reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, errors.Wrapf(err, "invalid object filesystem %q arguments", cephFilesystem.Name)
 	}
 
+	crushRoot := opconfig.GetCrushRootFromSpec(&cephCluster.Spec)
```

Comment:
Instead of getting the crushRoot from the cluster CR, we should expect that the `crushRoot` be set on all the pools. This is already what is expected when using a non-default crushRoot, and it will simplify this code.

Reply:
I thought about that too, but decided against it. I think overriding the CRUSH root for a pool is a different use-case than overriding the CRUSH root for the cluster.

Overriding the CRUSH root for a pool may put a pool in a specific availability zone for example. Changing the CRUSH root for the cluster should be mostly invisible and only be used to resolve conflicts between auto-created CRUSH labels and the root=default label.

So my argument would be that for the sake of user experience to keep this special handling and treat absent crushRoot in a pool to use the crush root form the cluster, because they serve different use-cases. When not setting a CRUSH root on a pool, it is expected that it uses the same as the cluster, IMO.

If we are to change this behaviour, I would like to be able to error-out if the crush root is not specified on the pool && the cluster uses a non-default crush root (otherwise, weird things will happen I’m afraid). This would again require looking up the crush root from the cluster, so there’s not much simplification either way.

Reply:
@horazont Agreed that it is a better UX if the default crush root is automatically applied to the pools, and would just be overridden by a setting in the pool specs. I suppose my question was also motivated by how many people will use this feature. If this will only affect one cluster I'd lean towards keeping the code simple and the UX is less important. If many clusters are affected, clearly the better UX is important. I'm ok with this change, just looking for ways to keep it simple.

Reply:
@travisn I went with the better UX. I don’t think we’d gain much by changing this in the code. Feel free to overrule (in which case I’ll change it).